### PR TITLE
Reduce test times

### PR DESCRIPTION
**Pull request by dirk-thomas**
http://ci.ros2.org/job/ros2_batch_ci_linux/406/
http://ci.ros2.org/job/ros2_batch_ci_osx/303/
http://ci.ros2.org/job/ros2_batch_ci_windows/452/

### CODE REVIEW

File: test_rclcpp/test/test_timer.cpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ TEST(CLASSNAME(test_time, RMW_IMPLEMENTATION), timer_fire_regularly) {
     printf("spin_node_some() for 4s\n");
     while (std::chrono::steady_clock::now() < start + 4.5 * period) {
       executor.spin_node_some(node);
-      std::this_thread::sleep_for(period / 25);
```
Comment (jacquelinekay) [inline test_rclcpp/test/test_timer.cpp]: Hmm, won't dividing by a smaller number actually have a larger result?

Comment (dirk-thomas) [inline test_rclcpp/test/test_timer.cpp]: Not really.

Before it was 1000 / 25 = 40ms.
Afterwards it is 100 / 10 = 10ms.

And usually sleeping less then 10ms doesn't make much sense anyway.

Comment (jacquelinekay) [inline test_rclcpp/test/test_timer.cpp]: Got it, I didn't see that you reduced the period.

Then you may want to change the print message `spin_node_some() for 4s` in line 59 to `400ms`.

File: test_communication/test/test_publisher_subscriber.cpp [hunk 2]
```diff
@@ -33,8 +33,8 @@ void publish(
   auto publisher = node->create_publisher<T>(
     std::string("test_message_") + message_type, custom_qos_profile);
 
-  rclcpp::WallRate time_between_cycles(1);
-  rclcpp::WallRate time_between_messages(10);
+  rclcpp::WallRate time_between_cycles(10);
```
Comment (jacquelinekay) [inline test_communication/test/test_publisher_subscriber.cpp]: While we're looking at this test, this variable has a misleading name. `time` implies it is some number of seconds, when actually it represents a rate.

File: test_rclcpp/test/test_timer.cpp (hunk 1 above)
Comment (dirk-thomas) [inline test_rclcpp/test/test_timer.cpp]: Good point. But who wants to maintain that string. Now it says "for some time" :wink:

File: test_communication/test/test_publisher_subscriber.cpp (hunk 2 above)
Comment (dirk-thomas) [inline test_communication/test/test_publisher_subscriber.cpp]: Good point. Addressed in a separate commit: 5d2532c50fcdef807cc77e305f23b910e4bf32b2

**Comment by jacquelinekay** [comment]
+1

**Comment by wjwwood** [comment]
lgtm

**Comment by esteve** [comment]
+1

**Pull request merged**